traffic_ops/traffic_ops_golang/deliveryservices_keys.go

```diff
@@ -34,10 +34,63 @@ import (
 	"github.com/apache/incubator-trafficcontrol/lib/go-log"
 	"github.com/apache/incubator-trafficcontrol/lib/go-tc"
 	"github.com/apache/incubator-trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/apache/incubator-trafficcontrol/traffic_ops/traffic_ops_golang/auth"
 	"github.com/basho/riak-go-client"
 	"github.com/jmoiron/sqlx"
 )
 
+type DeliveryServiceTenantInfo struct {
+	ID       int    `db:"id"`
+	XMLID    string `db:"xml_id"`
+	TenantID int    `db:"tenant_id"`
+}
+
+// returns true if the user has tenant access on this deliveryservice
+func (dsInfo DeliveryServiceTenantInfo) IsTenantAuthorized(user auth.CurrentUser, db *sqlx.DB) (bool, error) {
+	return auth.IsResourceAuthorizedToUser(dsInfo.TenantID, user, db)
+}
+
```

Comment:
tenancy checks for deliveryservices are needed outside of the `_keys` code.  We should discuss where this should go..